src/backend/parser/parse_utilcmd.c

```diff
@@ -1314,6 +1323,52 @@ transformTableLikeClause(CreateStmtContext *cxt, TableLikeClause *table_like_cla
 		list_free(parent_extstats);
 	}
 
+	/*
+	 * Likewise, copy indexes if requested
+	 * postgreSQL processes index statements after here in expandTableLikeClause(),
+	 * but we need indexes in transformDistributedBy() which is before expandTableLikeClause(),
+	 * So we both retain the index statements processing here and expandTableLikeClause.
+	 * the process here is just used by transformDistributedBy().
+	 */
+
+	if ((table_like_clause->options & CREATE_TABLE_LIKE_INDEXES) &&
+		relation->rd_rel->relhasindex)
+	{
+		List	   *parent_indexes;
+		ListCell   *l;
+
+		parent_indexes = RelationGetIndexList(relation);
+
+		foreach(l, parent_indexes)
+		{
+			Oid			parent_index_oid = lfirst_oid(l);
+			Relation	parent_index;
+			IndexStmt  *index_stmt;
+
+			parent_index = index_open(parent_index_oid, AccessShareLock);
+
+			/* Build CREATE INDEX statement to recreate the parent_index */
+			index_stmt = generateClonedIndexStmt(stmt->relation, parent_index,
+												 attmap, tupleDesc->natts, NULL);
+
+			/* Copy comment on index, if requested */
+			if (table_like_clause->options & CREATE_TABLE_LIKE_COMMENTS)
+			{
+				comment = GetComment(parent_index_oid, RelationRelationId, 0);
+
+				/*
+				 * We make use of IndexStmt's idxcomment option, so as not to
+				 * need to know now what name the index will have.
+				 */
+				index_stmt->idxcomment = comment;
+			}
+
+			/* Save it in the inh_indexes list for the time being */
+			cxt->inh_indexes = lappend(cxt->inh_indexes, index_stmt);
+
+			index_close(parent_index, AccessShareLock);
+		}
+	}
```

Comment:
Looks repeated with https://github.com/greenplum-db/gpdb/blob/59f4892874a272b4a98ea6a5138fdfbd2707f3b5/src/backend/parser/parse_utilcmd.c#L1293~L1332 , do we need to repeat one more time?

Reply:
After the merge, these code goes into `expandTableLikeClause()`. Can we reuse it?

Reply:
> can we choose distributed-by clause based on the pattern table?

We should have the same behavior as before.
if we choose distributed-by based on the pattern table, the distributed-by keys of the below SQL will be different from the main branch.

create table t1(tc1 int,tc2 int,tc2 int, primary key(tc2,tc3) distributed by tc3; --distributed key is tc3.
main branch:
create table t2(tc4 int, like t1 including indexes); --distributed key is chosen from unique indexes, (tc2,tc3)

after merge
create table t2(tc4 int, like t1 including indexes); --distributed key is tc3

Reply:
> Looks repeated with https://github.com/greenplum-db/gpdb/blob/59f4892874a272b4a98ea6a5138fdfbd2707f3b5/src/backend/parser/parse_utilcmd.c#L1293~L1332 , do we need to repeat one more time?

@zxuejing I just see what @Aegeaner 's comments, let me take a look based on that.

Reply:
Hi @zxuejing 

@Aegeaner ‘s concern makes sense to me. An idea from me is can we generate an extra `alter table set distributed by` statement?

The logic of Postgres is:

1.  generate a create table 
2. expand like clause (and the following stmts are just like `alter table`)

Ideas?

cc @adam8157 @Aegeaner